cmake/tests/cxx14_auto.cpp

```diff
@@ -0,0 +1,20 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2009 Andrey Semashev
+//  Copyright (c) 2015 Agustin Berge
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+template<typename T>
+auto check_f(T& i)
+{
+    T j = 1 + j;
+    return j;
+}
```

Comment:
I'm not opposed to this feature test. I'm opposed to using the results of this test to detect a NVCC problem related to a C++11 feature. Couldn't we just force HPX_WITH_CXX11_AUTO=OFF whenever HPX is configured with CUDA/NVCC enabled?